### PR TITLE
fix: port OSR code to new viz compositor codepath

### LINKED ISSUE #16983: Enable OSR in master and 5-0-x
It was disabled as a part of chromium 73 upgrade to avoid crashes with viz compositor.

### PR DESCRIPTION
**Pull request by brenca**
#### Description of Change

Fixes #16983.

There is an ongoing effort to upstream a large chunk of the patch to `chromium` (one of the guys from `CEF` is doing the upstreaming, and the `chromium` folks indicated that they are open to the patch).

TODO:
 - [x] Windows - Software rendering
 - [x] Windows - GPU rendering
 - [x] MacOS - Software rendering
 - [x] MacOS - GPU rendering
 - [x] Linux - Software rendering
 - [x] Linux - GPU rendering
 - [x] Backport to `5-0-x`

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed offscreen rendering not working with viz compositor.

### CODE REVIEW

File: atom/browser/osr/osr_host_display_client.cc [hunk 1]
```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2016 GitHub, Inc.
```
Comment (codebytere) [inline atom/browser/osr/osr_host_display_client.cc]: nit: update copyright year

**Comment by codebytere** [comment]
For a pull request this large, it's challenging to inspect every nook and cranny as thoroughly as i'd like but based on [this writeup](https://hackmd.io/Umx0zEPcRuWoIhV8KKEAHw?both) and my understanding of the code at hand this seems to be sufficient for us to merge. I don't see any glaring issues and it's (imo) more important to release this to consumers such that the beta cycle surfaces any bugs that may be present therein and that weren't caught in initial development.

**Comment by nornagon** [comment]
> one of the guys from CEF

btw, I'm pretty sure there's only one guy from CEF: Marshall Greenblatt :)

**Comment by brenca** [comment]
@nornagon Well, most of the viz osr work is actually done by Alexander Guettler, we have group PM with him, Marshall Greenblatt and @codebytere where we discussed how things are going. See https://bitbucket.org/chromiumembedded/cef/issues/2575

**Comment by jkleinsc** [comment]
@brenca can you rebase this PR?

**Pull request merged**